dd-java-agent/instrumentation/jdbc/src/main/java/datadog/trace/instrumentation/jdbc/AbstractPreparedStatementInstrumentation.java

```diff
@@ -64,21 +66,29 @@ public static AgentScope onEnter(@Advice.This final Statement statement) {
         return null;
       }
       try {
-        Connection connection = statement.getConnection();
-        DBQueryInfo queryInfo =
+        final Connection connection = statement.getConnection();
+        final DBQueryInfo queryInfo =
             InstrumentationContext.get(Statement.class, DBQueryInfo.class).get(statement);
         if (null == queryInfo) {
           logMissingQueryInfo(statement);
           return null;
         }
-
-        final AgentSpan span = startSpan(DATABASE_QUERY);
-        DECORATE.afterStart(span);
-        DBInfo dbInfo =
+        final AgentSpan span;
+        final DBInfo dbInfo =
             JDBCDecorator.parseDBInfo(
                 connection, InstrumentationContext.get(Connection.class, DBInfo.class));
+        final boolean injectTraceContext = DECORATE.shouldInjectTraceContext(dbInfo);
+
+        if (INJECT_COMMENT && injectTraceContext && DECORATE.isSqlServer(dbInfo)) {
+          final long spanID = DECORATE.setContextInfo(connection, dbInfo);
+          span = AgentTracer.get().buildSpan(DATABASE_QUERY).withSpanId(spanID).start();
```

Comment:
I know there is historically few comments in the tracer code, but I'd like to see that change :D

```suggestion
          // The span ID is pre-determined so that we can reference it when setting the context
          final long spanID = DECORATE.setContextInfo(connection, dbInfo);
          // we then force that pre-determined span ID for the span covering the actual query
          span = AgentTracer.get().buildSpan(DATABASE_QUERY).withSpanId(spanID).start();
```